### PR TITLE
GPI-2 new package

### PR DESCRIPTION
**Pull request by robert-mijakovic**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gpi-2/package.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+from spack import *
+
+
+def _verbs_dir():
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpi-2/package.py]: I think we used to have something like this in the OpenMPI package, but we removed it for several reasons.

File: var/spack/repos/builtin/packages/gpi-2/package.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    """GPI-2 is an API for the development of scalable, asynchronous and fault
+    tolerant parallel applications. It implements the GASPI specification
+    (www.gaspi.de)"""
+
+    homepage = "http://www.gpi-site.com"
+    url      = "https://github.com/cc-hpc-itwm/GPI-2/archive/v1.4.0.tar.gz"
+    git      = "https://github.com/cc-hpc-itwm/GPI-2.git"
+
+    version('master', branch='master')
+    version('next', branch='next')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpi-2/package.py]: This version will sort as older than all stable releases, we only make exceptions for `develop`, `master`, `head`, and `trunk`.

File: var/spack/repos/builtin/packages/gpi-2/package.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    version('1.4.0', sha256='3b8ffb45346b2fe56aaa7ba15a515e62f9dff45a28e6a014248e20094bbe50a1')
+
+    variant('mpi', default=False)
+    variant('fortran', default=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpi-2/package.py]: These variants need descriptions

File: var/spack/repos/builtin/packages/gpi-2/package.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        if not activated:
+            return '--without-{0}'.format(opt)
+        line = '--with-{0}'.format(opt)
+        path = _verbs_dir()
+        if (path is not None) and (path not in ('/usr', '/usr/local')):
+            line += '={0}'.format(path)
+        return line
+
+    @run_before('configure')
+    def autogen(self):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gpi-2/package.py]:
```suggestion
    def autoreconf(self):
```
Then you can get rid of `phases` above

File: var/spack/repos/builtin/packages/gpi-2/package.py (hunk 1 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: You are right, this one was based on OpenMPI. I have removed it as suggested!

File: var/spack/repos/builtin/packages/gpi-2/package.py (hunk 2 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: I'm not sure that I understood this one. You mean that next branch shouldn't be specified here or?

File: var/spack/repos/builtin/packages/gpi-2/package.py (hunk 3 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: I have overlooked this one. Thanks for noticing.

File: var/spack/repos/builtin/packages/gpi-2/package.py (hunk 4 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: Thanks I have cleaned this up.

File: var/spack/repos/builtin/packages/gpi-2/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: The branch must be specified, but you can name the version anything you want. I would recommend:
```suggestion
    version('master', branch='next')
```
if `next` is newer than stable releases.

Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: I think this is a good idea. I don't know what was developers intention but there are no commits to the master for quite some time.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: You can also do:
```python
version('develop', branch='next')
version('master', branch='master')
```
It really depends on which branches you plan on using, you don't have to add any at all if you don't care.

Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/gpi-2/package.py]: I don't know what is the plan of developers with the master branch and therefore I will return it and add develop version from your last suggestion.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**